app/javascript/admin/controllers/svg_icon_upload_controller.js

```diff
@@ -0,0 +1,62 @@
+import { Controller } from 'stimulus';
+
+export default class SvgIconUploadController extends Controller {
+  selectSvgIcon(event) {
+    const currentNavigationLinkModal = this.targets.scope.data.element.querySelector(
+      'div',
+    );
+
+    this.clearInvalidIconTypeMessage(currentNavigationLinkModal);
+
+    const svgIconContent = currentNavigationLinkModal.querySelector(
+      '#svg-icon-content',
+    );
+
+    const icon = event.target.files[0];
+    if (icon.type !== 'image/svg+xml') {
+      this.invalidIconTypeMessage(currentNavigationLinkModal, icon.type);
+      const navigationLinkId = currentNavigationLinkModal.getAttribute(
+        'nav-link-id',
+      );
+
+      const ableToRemoveSvgIconContent =
+        svgIconContent.value && !navigationLinkId;
+      if (ableToRemoveSvgIconContent) {
+        svgIconContent.value = null;
+        this.setSvgIconPreview(currentNavigationLinkModal, null);
+      }
+      return;
+    }
+
+    const reader = new FileReader();
+    reader.readAsText(icon);
+
+    reader.onload = (content) => {
+      const result = content.target.result;
+      svgIconContent.value = result;
+      this.setSvgIconPreview(currentNavigationLinkModal, result);
+    };
+  }
+
+  setSvgIconPreview(document, content) {
+    const iconPreview = document.querySelector('#svg-icon-preview');
```

Comment:
this `querySelector` is not needed if you use Stimulus targets

Reply:
Okay! Thanks a lot for the tip